### PR TITLE
Add `cropping_mode` to `DelayedAggregation`

### PR DESCRIPTION
**Pull request by bhigy**
Solve #104.

### CODE REVIEW

File: src/diart/blocks/aggregation.py [hunk 1]
```diff
@@ -130,10 +132,12 @@ def __init__(
         step: float,
         latency: Optional[float] = None,
         strategy: Literal["mean", "hamming", "first"] = "hamming",
+        cropping_mode: Literal["strict", "loose", "center"] = "loose"
     ):
         self.step = step
         self.latency = latency
         self.strategy = strategy
+        self.cropping_mode = cropping_mode
```
Comment (juanmc2005) [inline src/diart/blocks/aggregation.py:140]:
```suggestion
        assert cropping_mode in ["strict", "loose", "center"], f"Invalid cropping mode `{cropping_mode}`"
        self.cropping_mode = cropping_mode
```

File: src/diart/blocks/aggregation.py [hunk 2]
```diff
@@ -130,10 +132,12 @@ def __init__(
         step: float,
         latency: Optional[float] = None,
         strategy: Literal["mean", "hamming", "first"] = "hamming",
+        cropping_mode: Literal["strict", "loose", "center"] = "loose"
```
Comment (juanmc2005) [inline src/diart/blocks/aggregation.py:135]: There is also a lot of cropping going on in `AggregationStrategy` subclasses, we should add this parameter there as well so it remains consistent across all `DelayedAggregation` operations.

**Review by juanmc2005 — CHANGES_REQUESTED** [review]
Thanks for the fix! There are a couple of changes that are still needed before I can test  and merge.

File: src/diart/blocks/aggregation.py (hunk 2 above)
Comment (bhigy) [reply src/diart/blocks/aggregation.py:138]: Solved with e7d6dd7

File: src/diart/blocks/aggregation.py [hunk 3]
```diff
@@ -102,6 +102,8 @@ class DelayedAggregation:
         "mean": simple average
         "hamming": average weighted by the Hamming window values (aligned to the buffer)
         "any": no aggregation, pick the first overlapping window
+    cropping_mode: ("strict", "loose", "center"), optional
+        Defines the cropping mode. Defaults to "loose".
```
[the remark this answers is not in the document]
Comment (bhigy) [reply src/diart/blocks/aggregation.py:106]: Improved with e7d6dd7

File: src/diart/blocks/aggregation.py (hunk 2 above)
Comment (bhigy) [reply src/diart/blocks/aggregation.py:154]: Attempt to solve this in 4866586. Do you have a quick way of testing that?

Comment (juanmc2005) [reply src/diart/blocks/aggregation.py:154]: Yes I'll test it and get back to you, thanks!

**Review by juanmc2005 — APPROVED** [review]
I managed to reproduce the performance on AMI for latencies 0.5, 2.5 and 4. I think we're good to go

**Pull request merged**